src/widget-sources/widget-base-source.ts

```diff
@@ -184,19 +212,31 @@ export abstract class WidgetBaseSource<Props extends WidgetBaseSourceProps> {
       filterOwner,
       spatialFilter,
       spatialFiltersMode,
-      spatialFiltersResolution,
       abortController,
       operationExp,
+      viewState,
       ...params
     } = options;
     const {column, operation} = params;
+    const source = this.getModelSource(filterOwner);
+    
+    if (spatialFilter && source.spatialDataType !== 'geo') {
+      assert(
+        options.spatialFiltersResolution ?? viewState,
+        'spatialFiltersResolution or viewState is required for using spatialFilter with spatial indexes'
+      );
+    }
+
+    const spatialFiltersResolution = options.spatialFiltersResolution ?? (
+      getSpatialFiltersResolution({ source, viewState: viewState! })
+    );
```

Comment:
Ok, good call

Reply:
In the case the assert message would need to change to something like "viewState prop is required to compute automatic spatialFiltersResolution when using spatialFilter with spatial indexes. Either pass a `spatialFiltersResolution` prop or a `viewState` prop to avoid this error"